libraries/AC_Planck/AC_Planck.h

```diff
@@ -104,9 +112,22 @@ class AC_Planck {
     uint32_t timestamp_ms = 0;
   }_status;
 
+  enum
+  {
+    PLANCK_ACK,
+    PLANCK_NACK,
+    PLANCK_WAITING_FOR_ACK
+  } _ack_status = PLANCK_WAITING_FOR_ACK;
+  uint32_t _last_cmd_req_t_ms = 0;
+
   mavlink_channel_t _chan = MAVLINK_COMM_1;
 
   bool _was_at_location = false; //For debouncing at-location
 
   bool _is_status_ok(void) { return ((AP_HAL::millis() - _status.timestamp_ms) < 500); }
+  
+  void _sent_cmd_req() {
+    _last_cmd_req_t_ms = AP_HAL::millis();
+    _ack_status = PLANCK_WAITING_FOR_ACK;
```

Comment:
Does this handle several msg in a row ?

Reply:
yes; the `_ack_status` will just be reset each time.  APM is a single thread so theres no reentrancy or thread locking issues, but if APM incorrectly requests a command_req over and over again, its possible that there will be a mismatch between the ack and last command request sent.  But, the command request is only sent on new mode transitions or when starting a landing (in RTB)